### PR TITLE
Allow NodeHasMediaUse view filter to work on views with node relationships.

### PR DESCRIPTION
**Pull request by rosiel**
**GitHub Issue**: #1009 

* Other Relevant Links (Google Groups discussion, related pull requests,
 Release pull requests, etc.)

# What does this Pull Request do?

~~Very stupidly~~ Less naively, thanks to @adam-vessey! solves the issue I was having (see issue).

There's probably a better way, but I don't know how.

# What's new?
A in-depth description of the changes made by this PR. Technical details and
 possible side effects.

* Changes x feature to such that y
* Added x
* Removed y
* Does this change add any new dependencies? 
* Does this change require any other modifications to be made to the repository
 (i.e. Regeneration activity, etc.)? 
* Could this change impact execution of existing code?

# How should this be tested?

A description of what steps someone could take to:
* Reproduce the problem you are fixing (if applicable)
* Test that the Pull Request does what is intended.
* Please be as detailed as possible.
* Good testing instructions help get your PR completed faster.

# Documentation Status

* Does this change existing behaviour that's currently documented?
* Does this change require new pages or sections of documentation?
* Who does this need to be documented for?
* Associated documentation pull request(s): ___  or documentation issue ___

# Additional Notes:
Any additional information that you think would be helpful when reviewing this
 PR.

# Interested parties
Tag (@ mention) interested parties or, if unsure, @Islandora/committers

### CODE REVIEW

File: src/Plugin/views/filter/NodeHasMediaUse.php [hunk 1]
```diff
@@ -88,7 +88,7 @@ public function query() {
     $sub_query->join('media__field_media_of', 'of', 'm.mid = of.entity_id');
     $sub_query->fields('of', ['field_media_of_target_id'])
       ->condition('use.field_media_use_target_id', $term->id());
-    $this->query->addWhere(0, 'nid', $sub_query, $condition);
+    $this->query->addWhere(0, 'node_field_data.nid', $sub_query, $condition);
```
Comment (adam-vessey) [inline src/Plugin/views/filter/NodeHasMediaUse.php]: Not sure this should be hard-coded, as the table could be joined against multiple times? I'm thinking that the filter class could provide the appropriate alias, in some way?

Comment (adam-vessey) [reply src/Plugin/views/filter/NodeHasMediaUse.php]: Quick bit of poking, something like:

```suggestion
    /** @var \Drupal\views\Plugin\views\query\Sql $query */
    $query = $this->query;
    $alias = $query->ensureTable('node_field_data', $this->relationship);
    $query->addWhere(0, "{$alias}.nid", $sub_query, $condition);
```

Comment (adam-vessey) [reply src/Plugin/views/filter/NodeHasMediaUse.php]: Bit of other paranoia with the aliases here, but yeah, could pull in https://github.com/Islandora/islandora/compare/2.x...adam-vessey:islandora:fix/relationships ?

Comment (adam-vessey) [reply src/Plugin/views/filter/NodeHasMediaUse.php]: Might've gotten a little distracted, and thrown together something of a dependency-injection pass on the class as well: https://github.com/Islandora/islandora/compare/2.x...adam-vessey:islandora:fix/dependency-injection

Comment (rosiel) [reply src/Plugin/views/filter/NodeHasMediaUse.php]: oh dear. I screwed up and got a bunch of old commits. I'll try to rebase.

**Comment by aOelschlager** [comment]
I have tried testing this many times now and I keep getting errors when adding the member of relationship to the view. It could be an error on my part but I don't know.

Update: User error on my part pulling in the pr to test. I got a patch of the pr to work, thanks @rosiel!

**Review by aOelschlager — APPROVED** [review]
It works! I no longer see the database errors in the Drupal logs.

**Pull request merged**